tests/stub/driver_parameters/test_bookmark_manager.py

```diff
@@ -474,16 +474,182 @@ def test_should_resolve_database_name_with_system_bookmarks(self):
+            Neo4jBookmarkManagerConfig(
+                initial_bookmarks={
+                    "adb": adb_bookmarks
+                },
+                bookmarks_supplier=get_bookmarks
+            )
+        )
+
+        s1 = self._driver.session("w", database="neo4j")
+        tx1 = s1.begin_transaction({"return_bookmark": "bm1"})
+        list(tx1.run("RETURN 1 as n"))
+        tx1.commit()
+        s1.close()
+
+        s2 = self._driver.session("w")
+        tx2 = s2.begin_transaction({"return_bookmark": "bm2"})
+        list(tx2.run("RETURN 1 as n"))
+        tx2.commit()
+        s2.close()
+
+        self.assertEqual(4, len(get_bookmarks_calls))
+        self.assertEqual([
+            # routing table
+            ["system"],
+            # first tx
+            [None],
+            # name resolution
+            ["system"],
+            # second tx
+            [None]
+        ], get_bookmarks_calls)
+
+    def test_should_enrich_bookmarks_with_bookmark_supplier_result(self):
+        self._start_server(self._router, "router_with_db_name.script")
+        self._start_server(self._server, "transaction_chaining.script")
+
+        system_bookmarks = ["sys:bm3"]
+        neo4j_bookmarks = ["neo4j:bm3"]
+        adb_bookmarks = ["adb:bm1"]
+        bookmarks = {
+            "system": system_bookmarks,
+            "neo4j": neo4j_bookmarks
+        }
+
+        def get_bookmarks(db):
+            if db is None:
+                return system_bookmarks + neo4j_bookmarks
+            if db in bookmarks:
+                return bookmarks[db]
+            return []
+
+        self._driver = self._new_driver(
+            Neo4jBookmarkManagerConfig(
+                initial_bookmarks={
+                    "adb": adb_bookmarks
+                },
+                bookmarks_supplier=get_bookmarks
+            )
+        )
+
+        s1 = self._driver.session("w", database="neo4j")
+        tx1 = s1.begin_transaction({"return_bookmark": "bm1"})
+        list(tx1.run("RETURN 1 as n"))
+        tx1.commit()
+        s1.close()
+
+        s2 = self._driver.session("w")
+        tx2 = s2.begin_transaction({"return_bookmark": "bm2"})
+        list(tx2.run("RETURN 1 as n"))
+        tx2.commit()
+        s2.close()
+
+        self._server.reset()
+        begin_requests = self._server.get_requests("BEGIN")
+
+        self.assertEqual(len(begin_requests), 2)
+        self.assert_begin(
+            begin_requests[0],
+            bookmarks=system_bookmarks + neo4j_bookmarks + adb_bookmarks
+        )
+        self.assert_begin(
+            begin_requests[1],
+            bookmarks=["bm1"] + system_bookmarks + neo4j_bookmarks
+            + adb_bookmarks
+        )
+
+    def test_should_call_notify_bookmarks_when_new_bookmarks_arrive(self):
+        self._start_server(self._router, "router_with_db_name.script")
+        self._start_server(self._server, "transaction_chaining.script")
+
+        adb_bookmarks = ["adb:bm1"]
```

Comment:
Does this server any purpose in this test?

Is it to make sure the BMM doesn't call the consumer with the initially configured bookmarks?

Reply:
It was for checking the `adb_bookmarks` will not appear on `# first tx` and neither on the `# second tx`. Kind of a sanity check since it is already test in details in other tests related to the update method. s